Week 12: Movies

monkeybusiness101 opened the pull request:
Link to my deployed project:

https://wizardly-goldwasser-9c77ac.netlify.app/

My reflections on how this week's project turned out:

It was fun to work with APIs, useState, useEffect and newly introduced react Router.

Things I'd like to have clarified or explained in more detail:

How to handle the second fetch of the backdrop the returns undefined. I saw Maks lecture on Friday so need to dig into conditionals a bit more

I also have not looked into bad cases where users ends up on a site that does not exist for example

sandra-shumanteva-kaevska commented on code/public/index.html:
```diff
@@ -4,31 +4,15 @@
-    <title>Technigo React App</title>
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@700&display=swap" rel="stylesheet">
+    
+    <title>Movies</title>
```
Nice that you think of changing title :)

sandra-shumanteva-kaevska commented on code/src/BackButton.js:
```diff
@@ -0,0 +1,13 @@
+    return (
+        <div className="back-button">
+        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 30 30">
+            <path d="M27 14.5C27 7.596441 21.4035594 2 14.5 2S2 7.596441 2 14.5 7.5964406 27 14.5 27 27 21.403559 27 14.5zm-19.3388348-.353553l7.4852814-7.485282c.1952622-.195262.5118446-.195262.7071068 0l2.1213203 2.121321c.1952622.195262.1952622.511844 0 .707106L12.9644661 14.5l5.0104076 5.010408c.1952622.195262.1952622.511844 0 .707106l-2.1213203 2.121321c-.1952622.195262-.5118446.195262-.7071 ...
```
It is great that you have arrow button from "w3" 👍

sandra-shumanteva-kaevska commented on code/src/index.css:
```diff
@@ -1,13 +1,135 @@
+
+/* mobile view */
+
+@media only screen and (max-width: 400px) {
```
It is good that you think to make mobile responsive even it is not requirement 👍

sandra-shumanteva-kaevska reviewed (commented):
Nice structured, easy readable code and great naming of components. The web page is working properly without any bugs 👍  Great job.

 I think that you should just delete "console.log" in your code.

EmOlsz commented on code/src/Movies.js:
```diff
@@ -0,0 +1,41 @@
+        })
+        .then((text) => {
+            fetchMovies(text.results)
+            console.log(text.results)
```
👍

monkeybusiness101 replied:
fixed

EmOlsz commented on code/src/Movies.js:
```diff
@@ -0,0 +1,41 @@
+            {
+                movies.map(movie => (  
+                 <div className="movie">       
+                    <Link to={`${movie.id}`} key={movie.id}>
```
`key` attribute should be added to root element inside loop (so, parent of all elements inside loop). In your case, it's `<div className="movie"> `, not `<Link to={`${movie.id}`} key={movie.id}>`

monkeybusiness101 replied:
fixed

monkeybusiness101 commented on code/src/ShowMovie.js:
```diff
@@ -0,0 +1,54 @@
+    })
+    .then((text) => {
+        getDetails(text)
+        console.log(text)
```
fixed

monkeybusiness101 commented:
Hi @sandra-shumanteva-kaevska and @EmOlsz Thank you for the input and feedback!

I have removed console.log, moved the key to the parent and removed some spacing. Can you please have a look and approve my PR?

/Max
